### PR TITLE
Bp/ehf currency rework

### LINKED ISSUE #290: Downloading foreign currency invoices
When downloading foreign currency invoices, the amount that gets converted to local currency should go to the supplier/customer, and the difference from the actual paid amount from bank transaction should go to the motkonto result accountplan for that supplier/customer.

### LINKED ISSUE #304: Foreign currency when receiving Bank Transaction
Send invoice currency from bank transactions from FB, and display them in lodo bank.tabbankaccount.

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #290 
Closes #304

### CODE REVIEW

File: inc/javascript.inc.php [hunk 1]
```diff
@@ -268,39 +241,29 @@ function onCurrencyRateChange(element) {
  */
 function journalCurrencyChange(btn, action_url)
 {
-    var wrapper = btn.parentNode;
+    wrapper = btn.parentNode;
```
Comment (Baxxx) [inline inc/javascript.inc.php]: Investigate this

File: modules/journal/view/edit.php [hunk 2]
```diff
@@ -665,8 +676,8 @@
     <td align="right"><? print $_lib['format']->Amount($totalAmountIn) ?></td>
     <td align="right"><? print $_lib['format']->Amount($totalAmountOut) ?></td>
     <td colspan="2"></td>
-    <td align="right"><? print $totalForeignCurrency ." ". $_lib['format']->Amount($totalForeignAmountIn) ?></td>
-    <td align="right"><? print $totalForeignCurrency ." ". $_lib['format']->Amount($totalForeignAmountOut) ?></td>
+    <td align="right"><!-- <? print $totalForeignCurrency ." ". $_lib['format']->Amount($totalForeignAmountIn) ?> --></td>
```
Comment (Baxxx) [inline modules/journal/view/edit.php]: Remove this

File: modules/voucher/model/voucherinput.class.php [hunk 3]
```diff
@@ -223,8 +196,11 @@ function __construct($args) {
             $this->action['voucher_currency_update']      = true;
             unset($this->action['view_linedetails']);
         }
+        if(isset($args['action_postmotpost_save_currency'])) {
```
Comment (Baxxx) [inline modules/voucher/model/voucherinput.class.php:199]: There is no need for this. Use `$args['action_postmotpost_save_currency']` instead

File: modules/accounting/model/accounting.class.php [hunk 4]
```diff
@@ -1560,6 +1562,40 @@ private function update_voucher_line($fields, $VoucherID, $comment, $check_only_
         }
     }
 
+    // This function calculates one of: in/out amount, foreign amount or conversion rate, if we have the other two
+    // $in_or_out - string 'in' or 'out' that will set AmountIn or AmountOut if we are missing the amount.
+    //              Origin value of this field is set in vouchergui::currency2
+    function calculate_amount_foreign_and_rate(&$fields, $in_or_out = '') {
```
Comment (Baxxx) [inline modules/accounting/model/accounting.class.php:1568]: If we dont have two of the fields, give an error

File: modules/voucher/model/voucherinput.class.php [hunk 5]
```diff
@@ -663,7 +639,7 @@ function request($calledfrom) {
         global $_lib, $accounting;
         #Builds a has similar to request for db updates and the likes
         #N�r vi oppretter en ny linje, s� m� den f� v�re 0. M� gj�res til slutt da amountin/out nkan v�re satt fra kid ref s�k og oppslag
-        if($this->AmountIn == 0 && $this->AmountOut == 0 && !$this->action['voucherline_new'] && !$this->action['voucher_delete'] && !$this->action['voucher_head_delete'] && !$this->new) {
+        if($this->AmountIn == 0 && $this->AmountOut == 0 && !$this->action['voucherline_new'] && !$this->action['voucher_delete'] && !$this->action['voucher_head_delete'] && !$this->new && !($this->ForeignAmount && $this->ForeignConvRate)) {
```
Comment (Baxxx) [inline modules/voucher/model/voucherinput.class.php:642]: Investigate when does this happen.

**Comment by sls-lnluksa** [comment]
- [x] In open posts (postmotpost.list)
     - Bank transaction currency is not printed correctly (possible because of abs value)
     - Rename headers `Valuta inn	Valuta ut	Valuta/kurs` to `Valuta Valuta Kurs`
- [x] on bilagsutskrift (report.reskontrovoucherprint) 
     - Total sum on currency is not correct (should be accumulated as it is for NOK ). Signs on V.inn and V.out are not correct.
     - Remove MVA kode column
     - We don't print now columns invoice number, and KID. Print them between `Tekst	Matchet med`
     - Use translations for headers like `V.inn V.ut V.saldo` to `Valuta Valuta Saldo`
- [x] On journal page
     - when currency is selected print currency in/out fields.
     - if NOK is selected, do not print drop down but just as text.
     - Translate `Rate` -> `Kurs` 

Requested by: Hildegunn

File: modules/voucher/model/voucherinput.class.php (hunk 3 above)
Comment (Baxxx) [reply modules/voucher/model/voucherinput.class.php:199]: It is necessary after all.

**Comment by hildegunn95** [comment]
- [x] on bilagsutskrift (report.reskontrovoucherprint) Fakturano -> Fakturanr
- [x] on bilagsutskrift (report.reskontrovoucherprint) Do not print currency code in front of Saldo
- [x] postmotpost, align to right the 'Valuta', 'Valuta' and 'Kurs'
- [x] journal.edit currency field should be displayed for a new invoice(if you shall create one inv with 10 EUR)(EDIT: moved to a separate PR)
- [x] Display currency rate on journal.edit page with 4 decimal places(EDIT: moved to a separate PR)

File: modules/accounting/model/accounting.class.php [hunk 6]
```diff
@@ -1560,6 +1562,43 @@ private function update_voucher_line($fields, $VoucherID, $comment, $check_only_
...
+                $fields['voucher_ForeignAmount'] = $foreign_amount;
+            } else if(($amount == 0 || $amount == '') && $foreign_amount > 0 && $rate > 0) {
+                $amount = $foreign_amount * 100 / $rate;
+                if($in_or_out == 'in') {
+                    $fields['voucher_AmountIn'] = $amount;
+                } else if($in_or_out == 'out') {
+                    $fields['voucher_AmountOut'] = $amount;
+                }
+            } else if($fields['voucher_ForeignCurrencyID'] != '') {
+                $_lib['message']->add('Error: Too many fields are empty!');
```
Comment (stoivo) [inline modules/accounting/model/accounting.class.php]: Error: Too many fields are empty!
Feil:  For mange felt er tomme!

File: inc/javascript.inc.php (hunk 1 above)
Comment (Baxxx) [reply inc/javascript.inc.php]: Revert this

**Review by stoivo — APPROVED** [review]
LGTM

**Pull request merged**